### PR TITLE
Fix create database conflicts with name in schema

### LINKED ISSUE #4566: Can't create a database that shares a name with a standard library function
```
edgedb> create database range;
edgedb error: SchemaError: function 'std::range(lower: OPTIONAL std::anypoint={}, upper: OPTIONAL std::anypoint={}, NAMED ONLY inc_lower: std::bool=true, NAMED ONLY inc_upper: std::bool=false, NAMED ONLY empty: std::bool=false)' already exists
```

### PR DESCRIPTION
**Pull request by aljazerzen**
Closes #4566

### CODE REVIEW

**Comment by aljazerzen** [comment]
Why does "CreateDatabase" even go through DDL? Does it needs to compute a delta? To be applied to the global schema?

It seems to me that all `DatabaseCommand`s should be simple operations - at least delta-wise.

**Comment by msullivan** [comment]
> Why does "CreateDatabase" even go through DDL? Does it needs to compute a delta? To be applied to the global schema?

I believe that's right, yeah.

File: tests/test_database.py [hunk 1]
```diff
@@ -76,6 +76,15 @@ async def test_database_create_03(self):
         finally:
             await tb.drop_db(self.con, 'databasename')
 
+    async def test_database_create_04(self):
+        if not self.has_create_database:
+            self.skipTest("create database is not supported by the backend")
+
+        # create database name that conflicts with names in schema
+        await self.con.execute('CREATE DATABASE range;')
```
Comment (msullivan) [inline tests/test_database.py:84]: Could we do some testing that having this name collision doesn't lead to any trouble?

Comment (aljazerzen) [reply tests/test_database.py:84]: Like what? Write to the new database?

Comment (msullivan) [reply tests/test_database.py:84]: No, like using the objects that it collided with

**Pull request merged**